include/ginkgo/core/base/types.hpp

```diff
@@ -91,6 +93,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif  // defined(_WIN32) || defined(__CYGWIN__)
 
 
+/**
+ * Encapsulates all arguments in one macro, so a comma inside an argument (e.g.
+ * std::tuple<int, float>) still works as an argument and is not considered as
+ * multiple arguments.
+ */
+#define GKO_UNPACK(...) __VA_ARGS__
```

Comment:
You can also do this without an additional macro by adding parentheses around the parameter:
`GKO_MACRO((parameter, with, commas), (next, parameter))`

Reply:
I actually did not know that, I thought I need such a macro. I will remove it and see if the next PR (the CB-GMRES) can work with your suggestion.